File: api/src/main/java/put/poznan/sport/entity/SportFacilityNews.java
package put.poznan.sport.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Builder
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
public class SportFacilityNews {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private Integer sportFacilitiesId;
    private String title;
    private String description;
    private String imageUrl;

    @ManyToOne
    @JoinColumn(name = "sportFacilitiesId", insertable = false, updatable = false)
    @JsonIgnore
    private SportFacility sportFacility;
}

File: mobile_app/lib/models/details/sport_facility_news.dart
class SportFacilityNews {
  int? id;
  int? sportFacilitiesId;
  String? title;
  String? description;
  String? imageUrl;

  SportFacilityNews({
    this.id,
    this.sportFacilitiesId,
    this.title,
    this.description,
    this.imageUrl,
  });

  factory SportFacilityNews.fromJson(Map<String, dynamic> json) {
    return SportFacilityNews(
      id: json['id'] as int?,
      sportFacilitiesId: json['sportFacilitiesId'] as int?,
      title: json['title'] as String?,
      description: json['description'] as String?,
      imageUrl: json['imageUrl'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['sportFacilitiesId'] = sportFacilitiesId;
    data['title'] = title;
    data['description'] = description;
    data['imageUrl'] = imageUrl;

    return data;
  }
}
